### PR TITLE
Create Upgrading_CoreDNS.md

### PR DESCRIPTION
**Pull request by chrisohaver**
This doc covers an "Upgrade Strategy" for a k8s admin who is upgrading CoreDNS.

### CODE REVIEW

File: kubernetes/Upgrading_CoreDNS.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+
+# Handling CoreDNS Upgrades in Kubernetes Clusters
+
+When upgrading a Kubernetes cluster, some care should be taken when upgrading the CoreDNS component to avoid backward incompatible configuration failures. At a high level, you'll want to review the CoreDNS release notes to see if the new version introduces backward incompatibilities or deprecations and adjust your Corefile as needed before upgrading CoreDNS.
+
+
+## Identifying Backward Incompatibilities and Deprecations
+
+If you are upgrading CoreDNS as part of a Kubernetes upgrade, backward incompatibilities should be noted in the Kubernetes release notes of each version. You may also review the CoreDNS release notes as described below.
```
Comment (rajansandeep) [inline kubernetes/Upgrading_CoreDNS.md]: Are there any backward incompatible changes mentioned in the Kubernetes releases yet? 

K8s release notes https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md
Maybe we can add this link as well.

So we'll need to mention backward incompatible changes on the Kubernetes side while writing the release notes for PRs updating the CoreDNS versions.

Comment (chrisohaver) [reply kubernetes/Upgrading_CoreDNS.md]: It means we would have to maintain it, but i think it's a good idea.

If we want to do that, I think it should be in a separate document.

File: kubernetes/Upgrading_CoreDNS.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+When upgrading a Kubernetes cluster, some care should be taken when upgrading the CoreDNS component to avoid backward incompatible configuration failures. At a high level, you'll want to review the CoreDNS release notes to see if the new version introduces backward incompatibilities or deprecations and adjust your Corefile as needed before upgrading CoreDNS.
+
+
+## Identifying Backward Incompatibilities and Deprecations
+
+To identify possible backward incompatibilities, you'll need to review the CoreDNS release notes.  CoreDNS Release Notes are located in the [CoreDNS blog](https://coredns.io/blog/). The CoreDNS deprecation policy is such that we will only introduce backward incompatibilities in x.x.0 and x.x.1 releases.  So, for example, if you are upgrading from 1.1.5, to 1.3.1, you should check release notes for 1.2.0, 1.2.1, 1.3.0, and 1.3.1 for any deprecation/backward incompatibility notices.
+
+If you’ve discovered any backward incompatibility notices, you should review your Corefile to see if you are affected.
+
+
```
Comment (fturib) [inline kubernetes/Upgrading_CoreDNS.md]: Should we add a paragraph to show how to make it when downtime is not so big problem:
- upgrade your deployment manually ? or through tool ?
- edit configmap to tune the config
- verify coredns starts with checking pods and logs

File: kubernetes/Upgrading_CoreDNS.md (hunk 1 above)
Comment (rajansandeep) [reply kubernetes/Upgrading_CoreDNS.md]: sgtm, I will start a new doc on this. 
Maybe link that to this document later.

File: kubernetes/Upgrading_CoreDNS.md (hunk 2 above)
Comment (chrisohaver) [reply kubernetes/Upgrading_CoreDNS.md]: @fturib, adjusted per your comments, PTAL.

**Pull request merged**